src/wrapper/java_vm.rs

```diff
@@ -0,0 +1,87 @@
+use errors::*;
+use JNIEnv;
+
+use sys;
+
+use std::ptr;
+use std::ops::Deref;
+
+/// The invocation API.
+pub struct JavaVM(*mut sys::JavaVM);
+
+unsafe impl Send for JavaVM {}
+unsafe impl Sync for JavaVM {}
+
+impl JavaVM {
+    /// Create a JavaVM from a raw pointer.
+    pub fn from_raw(ptr: *mut sys::JavaVM) -> Result<Self> {
```

Comment:
This should probably be marked unsafe since, even if the pointer is non-null, it could still be otherwise invalid.